app/src/main/java/com/github/onlynotesswent/model/folder/FolderRepositoryFirestore.kt

```diff
@@ -199,39 +350,120 @@ class FolderRepositoryFirestore(private val db: FirebaseFirestore) : FolderRepos
     }
   }
 
-  override fun deleteFolderContents(
+  override suspend fun deleteFolderContents(
       folder: Folder,
       noteViewModel: NoteViewModel,
       onSuccess: () -> Unit,
-      onFailure: (Exception) -> Unit
+      onFailure: (Exception) -> Unit,
+      useCache: Boolean
   ) {
+    // Update the cache if needed
+    if (useCache) {
+      getSubFoldersOf(
+          parentFolderId = folder.id,
+          onSuccess = { subFolders ->
+            subFolders.forEach { subFolder ->
+              CoroutineScope(Dispatchers.IO).launch {
```

Comment:
Best to use withContext I believe, gives more control in case the englobing coroutine fails

Reply:
`withContext` does not work in this case unfortunately. I had to use `CoroutineScope` instead :'(
The error is the following: `Suspension functions can be called only within coroutine body`